### PR TITLE
Update AuthenticateUser

### PR DESCRIPTION
**Pull request by kimlisa**
- implemented and unit tested matchEmailAndPassword db utility
- modified AuthenticateUser:
  - removed grabbing/validiating UUID from both service and unit test
  - uses matchEmailAndPassword utility
- linted and formatted

### CODE REVIEW

File: service/db.go [hunk 1]
```diff
@@ -708,3 +708,76 @@ func deleteEmailTokenRow(uuid string) error {
...
+	}
+
+	if err := validatePassword(password); err != nil {
+		return nil, err
+	}
+
+	command := `SELECT uuid, first_name, last_name, email, organization, 
+       				created_timestamp, is_verified, password, permission_level, prospective_email
+				FROM user_svc.accounts 
+				WHERE email = $1
```
Comment (faraonc) [inline service/db.go]: include `password =  $2`, we are using an indices

Comment (faraonc) [reply service/db.go]: you will not have to `comparePassword` below

Comment (kimlisa) [reply service/db.go]: we store/retrieve hashed passwords from db, unless the hash will always generate the same string value if i hash user unhashed password??

Comment (faraonc) [reply service/db.go]: I wonder if we can hash it before sending the query in that way the service is not doing work, and we are utilizing the indices in our db, so we can get rid of this func.

```
// comparePassword compares hashedPassword retrieved from DB and the password from User request.
// Returns nil if match, error if not match or error from bcrypt.
func comparePassword(hashedPassword string, password string) error {
	if hashedPassword == "" || password == "" {
		return consts.ErrInvalidPassword
	}

	err := bcrypt.CompareHashAndPassword([]byte(hashedPassword), []byte(password))
	if err != nil {
		return err
	}

	return nil
}
```

Comment (faraonc) [reply service/db.go]: package we are using does not do implement that functionality. It is maybe because of security.

**Pull request merged**